pkg/bootstrap/platform/discovery_test.go

```diff
@@ -0,0 +1,78 @@
+// Copyright Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package platform
+
+import (
+	"log"
+	"os"
+	"reflect"
+	"testing"
+	"time"
+)
+
+func UnsetAllEnv(t *testing.T) {
+	err := os.Unsetenv("GCP_METADATA")
+	if err != nil {
+		t.Errorf("Unable to unset GCP_METADATA: %v", err)
+	}
+	// in diff func because other plats might use env vars.
+}
+
+func TestDiscoverWithTimeout(t *testing.T) {
+	tests := []struct {
+		desc       string
+		timeout    time.Duration
+		platKey    string
+		platVal    string
+		platExpect Environment
```

Comment:
Instead of taking a full platform here, I suggest taking a metadata func. Something like:

```golang
type metadataFn func(map[string]string) bool
...
func isGCPMetadata(m map[string]string) bool {
  if proj , ok := m[platform.GCPProject]; ok {
     return proj == "FOO"
  }
  ...
  return false
}
...
{ "gcp", 1 * time.Second, "GCP_METADATA", "FOO|BAR|BAZ|MAR", isGCPMetadata()}
...
got := DiscoverWithTimeout(tt.timeout).Metadata()
```

That way, you don't have to expose details like the env structs that aren't already exposed.

Reply:
ah got it thanks!

Reply:
addressed.